src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -483,7 +499,12 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     if (restartVideo) {
       this.logger.info('restarting local video to switch to new device');
       this.boundAudioVideoController.restartLocalVideo(() => {
-        this.releaseMediaStream(oldStream);
+        // tracks of oldStream should be stopped when video tile is disconnected from MediaStream
+        // otherwise, camera is still being accessed and we need to stop it here.
+        if (oldStream && oldStream.active) {
+          this.logger.error('previous media stream is not stopped during restart video');
```

Comment:
I am being really cautious, `restartLocalVideo()` stops `oldStream`, and this callback of `releaseStream` should not be needed as mentioned in the comment.
When it happens, it indicates an implementation error, it's possible from us or it could be from a wrong API calling sequence. 
I am okay with just logging a warning.